datafusion/sqllogictest/test_files/scalar.slt

```diff
@@ -1940,7 +1940,7 @@ select position('' in '')
 ----
 1
 
-query error DataFusion error: Error during planning: Error during planning: Int64 and Int64 are not coercible to a common string
+query error DataFusion error: Error during planning: Error during planning: The signature expected NativeType::String but received Int64
```

Comment:
I think the error message could be confusing 🤔 . I'm not sure. If providing both of them are `NativeType`, maybe better? 
```suggestion
query error DataFusion error: Error during planning: Error during planning: The signature expected NativeType::String but received NativeType::Int64
```
or
```suggestion
query error DataFusion error: Error during planning: Error during planning: The signature expected NativeType::String but received ArrowType Int64
```
?